### PR TITLE
[color spaces] Update specs for color.mix() and color.complement()

### PR DESCRIPTION
**Pull request by dvdherron**
## Description
Updated tests for [color.mix()](https://github.com/sass/sass/blob/main/proposal/color-4-new-spaces.md#colormix-1) and [color.complement()](https://github.com/sass/sass/blob/main/proposal/color-4-new-spaces.md#colormix-1).

See https://github.com/sass/sass/issues/2831

### CODE REVIEW

File: spec/core_functions/color/complement.hrx [hunk 1]
```diff
@@ -88,8 +88,8 @@ Error: Missing argument $color.
   |       ^^^^^^^^^^^^ invocation
   '
   ,--> sass:color
-1 | @function complement($color) {
-  |           ================== declaration
+1 | @function complement($color, $space: null) {
```
Comment (dvdherron) [inline spec/core_functions/color/complement.hrx:158]: @mirisuzanne For functions like this that don't have similar wpt tests, is there a way to test what values they return now?

Comment (mirisuzanne) [reply spec/core_functions/color/complement.hrx:158]: @dvdherron we could use something like colorjs.io to mimic the function logic, and take results from that. It doesn't have the same function, but  it should have all the features we would need. (I used it sometimes while writing the spec).

I've also mocked up some functions (and tests) in https://github.com/oddbird/sass-colors-sample which I forgot about. I'll add links to both in the trello card.

**Comment by dvdherron** [comment]
@stacyk and/or @mirisuzanne This is ready for an internal review. I've updated the color.mix() and color.complement() functions with the new color space functionality. 

- I know we are still figuring out a lot about what these tests will look like exactly. I'm mainly looking to hear if you think there are more things I should consider testing. Is it a good blend of color spaces? 
- I made some simple steps to mimic the way I understand these functions should work on colorjs. I'll share those in the Trello card if you also want to test the outputs.
- color.mix() was originally one file but once the tests got past ~500 lines the linter split make `mix` a directory and turned each virtual file into actual files. 
- It's not clear to me exactly what format Sass will return for certain results. Are hex codes prefered? Rgb functions (percents or comma-separated numbers?).

**Review by mirisuzanne — APPROVED** [review]
@dvdherron I would have to ask Natalie most of these questions as well, so I'm not sure I have good answers. For output, the expectation is that the color will be returned in the same space as it was originally written (for color.mix, we use the first color's space). But I'm not sure how Natalie plans to handle that in legacy spaces - we left that up to implementations. I think at this point Sass uses a color name if it's a perfect match, otherwise hex if it's an opaque color, and comma-separated `rgba()` otherwise.

**Pull request closed without merging**